base/services/telnetd/telnetd.c

```diff
@@ -367,12 +367,20 @@ static void RunShell(client_t *client)
    STARTUPINFO           si;
    PROCESS_INFORMATION   piProcInfo;
    SECURITY_ATTRIBUTES   saAttr;
+   char cmd_path[MAX_PATH];
+
+   if (!GetEnvironmentVariableA("COMSPEC", cmd_path, ARRAYSIZE(cmd_path)))
+   {
+      if (GetSystemDirectoryA(cmd_path, ARRAYSIZE(cmd_path)))
+      {
+         StringCchCatA(cmd_path, ARRAYSIZE(cmd_path), "\\cmd.exe");
+      }
+      else
+      {
+         StringCchCopyA(cmd_path, ARRAYSIZE(cmd_path), "C:\\ReactOS\\system32\\cmd.exe");
+      }
+   }
 
-   const char *name = "c:\\reactos\\system32\\cmd.exe";
```

Comment:
Should ROS have a global `#define DEFAULT_SYSTEM32_PATH "C:\\ReactOS\\System32"` (or static var, if that would apply), to use where needed? Easier to maintain/read/search/...

Reply:
Which system header will contain it?